custom/ewsghana/reports/specific_reports/stock_status_report.py

```diff
@@ -0,0 +1,324 @@
+                            datalist.append([row['product_code'], row['with_stock']])
+                    ret_data.append({'color': chart_config['label_color'][k], 'label': k, 'data': datalist})
+                return ret_data
+
+            chart = MultiBarChart('', x_axis=Axis('Products'), y_axis=Axis(''))
+            chart.rotateLabels = -45
+            chart.marginBottom = 120
+            chart.stacked = False
+            for row in convert_product_data_to_stack_chart(product_availability, self.chart_config):
+                chart.add_dataset(row['label'], [
+                    {'x': r[0], 'y': r[1]}
+                    for r in sorted(row['data'], key=lambda x: x[0])], color=row['color']
+                )
+            return [chart]
+        return []
+
+
+class MonthOfStockProduct(EWSData):
+
+    slug = 'mos_product'
+    title = 'Current MOS by Product'
+    show_chart = False
+    show_table = True
+    use_datatables = True
+
+    @property
+    def headers(self):
+        headers = DataTablesHeader(*[DataTablesColumn('Location')])
+
+        for product in get_products(self.config['program'],
+                                    self.config['products'],
+                                    self.config['domain']).order_by('code'):
+            headers.add_column(DataTablesColumn(product.code))
+
+        return headers
+
+    @property
+    def rows(self):
+        products = get_products(self.config['program'],
+                                self.config['products'],
+                                self.config['domain']).order_by('code')
+        rows = []
+        if self.config['location_id']:
+            supply_points = SQLLocation.objects.filter(parent__location_id=self.config['location_id'])\
+                .order_by('name').exclude(supply_point_id__isnull=True)
+            for sp in supply_points:
+                if sp.location_type == 'facility':
+                    cls = StockLevelsReport
+                else:
+                    cls = StockStatus
+                url = make_url(
+                    cls,
+                    self.config['domain'],
+                    '?location_id=%s&filter_by_program=%s&startdate=%s'
+                    '&enddate=%s&report_type=%s&filter_by_product=%s',
+                    (sp.location_id, self.config['program'], self.config['startdate'],
+                    self.config['enddate'], self.config['report_type'],
+                    '&filter_by_product='.join(self.config['products'])))
+
+                row = [link_format(sp.name, url)]
+                for p in products:
+                    stock = StockState.objects.filter(sql_product=p, case_id=sp.supply_point_id)\
+                        .order_by('-last_modified_date')
+
+                    if stock:
+                        if stock[0].get_monthly_consumption():
+                            row.append(int(stock[0].stock_on_hand / stock[0].get_monthly_consumption()))
```

Comment:
if `get_monthly_consumption()` has to compute the default value it can be an expensive call, should set a variable then check that and reuse it if you need to.